optimade/models/tests/test_models.py

```diff
@@ -44,6 +44,12 @@ def test_bad_structures(self):
             ):
                 StructureResource(**StructureMapper.map_back(structure))
 
+    def test_simple_relationships(self):
```

Comment:
I think we should test the validators as well, e.g. that passing a wrong `type` raises a specific error?

Reply:
I've just done this, but had to modify `jsonapi` to use `ValueError`'s as the assertion errors were causing all future validators to fail... if I get time (lol) I'll create a MWE and send it to pydantic

Reply:
> (...) if I get time (lol) I'll create a MWE and send it to pydantic

I guess `fastapi` just needs to update to `pydantic` v1.x